lib/Backend/Inline.cpp

```diff
@@ -3273,13 +3316,25 @@ Inline::InlineCallTarget(IR::Instr *callInstr, const FunctionJITTimeInfo* inline
 
     explicitThisArgOut->ReplaceSrc2(startCall->GetDst());
 
+    if (isCallInstanceFunction)
+    {
+        IR::Instr * functionArg = secondArgOut;
+        IR::Instr * bytecodeArgOutUse = IR::Instr::New(Js::OpCode::BytecodeArgOutUse, callInstr->m_func);
+        callInstr->ReplaceSrc1(functionArg->GetSrc1());
+        bytecodeArgOutUse->SetSrc1(functionArg->GetSrc1());
+        callInstr->InsertBefore(bytecodeArgOutUse); // Need to keep the function argout live till the call instruction for it to be captured by any bailout in the call sequence.
+        functionArg->Remove();
+    }
+
     IR::Instr * bytecodeArgOutUse = IR::Instr::New(Js::OpCode::BytecodeArgOutUse, callInstr->m_func);
```

Comment:
Is this deliberately duplicated from inside the above conditional? Should it be declared prior to the conditional and conditionally configured, or do we actually want both?

Reply:
Ah, I see now that they differ on the argument (argument 1 vs argument 2).